test/matchOptions.html

```diff
@@ -0,0 +1,84 @@
+<!-- Max Tarlov 10/19/19 testing match constraint options -->
+<html>
+	<title>matchMax Constraints Test</title>
+	<head>
+		<link rel="stylesheet" type="text/css" href="../spot.css">
+		<script src="../build/spot.js"></script>
+		<script src="../build/constraints.js"></script>
+		<script src="../candidategenerator.js"></script>
+	</head>
+	<body>
+		<pre id="results-container"></pre>
+
+		<script>
+			var sTree = {
+				cat: "cp",
+				id: "root",
+				children: [
+					{
+						cat: "xp",
+						id:"xp1",
+						children: [
+							{
+								cat:"x0",
+								id: "one"
+							}
+						]
+					},
+					{
+						cat: "xp",
+						id: "xp2",
+						children: [
+							{
+								cat: "xp",
+								id:"xp3",
+								func: true,
+								children:[
+									{
+										cat: "x0",
+										id: "two-func"
+									}
+								]
+							},
+							{
+								cat: "xp",
+								id: "xp4",
+								silentHead: true,
+								children: [
+									{
+										cat:"x0",
+										id: "three-silent"
+									}
+								]
+							}
+						]
+					}
+				]
+			}
+			/*
+			function maxMax(stree, ptree, cat){
+				return matchSP(stree, ptree, cat, {maxSyntax: true, maxProsody: true});
+			}
+			function maxMin(stree, ptree, cat){
+				return matchSP(stree, ptree, cat, {maxSyntax: true, minProsody: true});
+			}
+			function minMin(stree, ptree, cat){
+				return matchSP(stree, ptree, cat, {minSyntax: true, minProsody: true});
+			}
+			function minMax(stree, ptree, cat){
+				return matchSP(stree, ptree, cat, {minSyntax: true, maxProsody: true});
+			}
+			*/
+
+
+			//'matchPS-phi-{"maxProsody":true}'
+			var con = ['matchSP-xp', 'matchSP-xp-{"maxSyntax":true}', 'matchSP-xp-{"maxProsody":true}', 'matchSP-xp-{"requireOvertHead":true}', 'matchPS-phi-{"requireLexical":true}'];
```

Comment:
matchSP-xp-{"maxProsody":true} and matchPS-phi-{"requireLexical":true} are a bit strange as constraints go, so I'm having trouble deciding if they are "working". Please sanity-check my interpretations:
* matchSP-xp-{"maxProsody":true} = assign a violation for every xp that doesn't have a corresponding maximal phi. --> That does seem to be happening.
* matchPS-phi-{"requireLexical":true} = assign a violation for every phi that doesn't have a corresponding lexical XP. --> that also seems to be happening.

Reply:
these are the definitions I was working with